### ISSUE TITLE
Suggestion: Change `ICommandHandler` type to return `TryAsync` instead of `TryOptionAsync`

### ISSUE THREAD

**Issue by seangwright** [KyleMcMaster/payroll-processor#46]
The `TryOptionAsync` type is good at representing an async operation that can result in Success/Fail and in the Success case there is a Some/None value.

This makes sense for our queries because the query itself could succeed or fail (throw exception) and the query itself could return the item(s) or nothing, if they were not found.

However, commands typically don't return anything (except maybe an identifier from a completed operation) and they aren't necessarily looking for anything in particular. If an item that is needed for a command can't be found, that command is probably best considered to be failed.

In summary, we should consider changing `ICommandHandler<TCommand>` to have a `Execute()` method with the signature

```csharp
TryAsync<Unit> Execute(TCommand command, CancellationToken token);
```
